File: output/llama3.1-8b/avatar/Python/Java/atcoder_AGC027_B.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.Arrays;

public class atcoder_AGC027_B {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new FileReader("input.txt"));
        String[] input = br.readLine().split("\\s+");
        int N = Integer.parseInt(input[0]);
        int X = Integer.parseInt(input[1]);
        int[] A = Arrays.stream(input, 2, input.length)
                .mapToInt(Integer::parseInt).toArray();

        long[] S = new long[N];
        long sum = 0;
        for (int i = 0; i < N; i++) {
            sum += A[i];
            S[i] = sum;
        }

        long ans = Long.MAX_VALUE;
        for (int k = 1; k <= N / 2 + 1; k++) {
            int start = Math.max(0, N - 2 * k - 1);
            long E = k * X + 2 * Arrays.stream(S, start, N)
                    .mapToLong(i -> i).sum();
            ans = Math.min(ans, E);
        }

        System.out.println(ans + N * X + 5 * S[S.length - 1]);
    }
}

File: output/llama3.1-8b/avatar/Java/Python/atcoder_AGC027_B.py
import sys

class atcoder_AGC027_B:
    INF = sys.maxsize

    def solve(self):
        sc = input()
        N, X = map(int, sc.split())
        x = [int(i) for i in input().split()]
        xsum = [0] * (N + 1)
        for i in range(N):
            x[i] = int(x[i])
            xsum[i+1] = xsum[i] + x[i]

        ans = X * N + 5 * sum(x)

        for i in range(1, N):
            cost = X * i + 5 * (sum(x) - sum(x[:N-i]))
            j, k = 5, N - i
            while j <= 10 and k >= 0:
                if cost > ans:
                    break
                cost += j * (xsum[k] - xsum[max(k - i, 0)])
                j += 2
                k -= i
            ans = min(ans, cost)

        print(ans + N * X)

atcoder_AGC027_B().solve()